src/category_theory/bicategory/functor.lean

```diff
@@ -212,10 +234,196 @@ def comp : oplax_functor B D :=
   { dsimp,
     simp only [map₂_right_unitor, map₂_comp, map_comp_naturality_right_assoc,
       whisker_left_comp, assoc] },
-  .. F.to_prelax_functor.comp G.to_prelax_functor }
+  .. (F : prelax_functor B C).comp ↑G }
+
+/--
+A structure on an oplax functor that promotes an oplax functors to a pseudofunctor.
+See `pseudofunctor.mk_of_oplax`.
+-/
+@[nolint has_inhabited_instance]
+structure pseudo_core (F : oplax_functor B C) :=
+(map_id_iso (a : B) : F.map (𝟙 a) ≅ 𝟙 (F.obj a))
+(map_comp_iso {a b c : B} (f : a ⟶ b) (g : b ⟶ c) : F.map (f ≫ g) ≅ F.map f ≫ F.map g)
+(map_id_iso_hom' : ∀ {a : B}, (map_id_iso a).hom = F.map_id a . obviously)
+(map_comp_iso_hom' : ∀ {a b c : B} (f : a ⟶ b) (g : b ⟶ c),
+  (map_comp_iso f g).hom = F.map_comp f g . obviously)
+
+restate_axiom pseudo_core.map_id_iso_hom'
+restate_axiom pseudo_core.map_comp_iso_hom'
+attribute [simp] pseudo_core.map_id_iso_hom pseudo_core.map_comp_iso_hom
+
+end
 
 end oplax_functor
 
+/--
+This auxiliary definition states that pseudofunctors preserve the associators
+modulo some adjustments of domains and codomains of 2-morphisms.
+-/
+/-
+We use this auxiliary definition instead of writing it directly in the definition
+of pseudofunctors because doing so will cause a timeout.
+-/
+@[simp]
+def pseudofunctor.map₂_associator_aux
+  (obj : B → C) (map : Π {X Y : B}, (X ⟶ Y) → (obj X ⟶ obj Y))
+  (map₂ : Π {a b : B} {f g : a ⟶ b}, (f ⟶ g) → (map f ⟶ map g))
+  (map_comp : Π {a b c : B} (f : a ⟶ b) (g : b ⟶ c), map (f ≫ g) ≅ map f ≫ map g)
+  {a b c d : B} (f : a ⟶ b) (g : b ⟶ c) (h : c ⟶ d) : Prop :=
+map₂ (α_ f g h).hom = (map_comp (f ≫ g) h).hom ≫ ((map_comp f g).hom ▷ map h) ≫
+  (α_ (map f) (map g) (map h)).hom ≫ (map f ◁ (map_comp g h).inv) ≫ (map_comp f (g ≫ h)).inv
+
+/--
+A pseudofunctors `F` between bicategories `B` and `C` consists of functions between objects,
+1-morphisms, and 2-morphisms.
+
+Unlike functors between categories, functions between 1-morphisms do not need to strictly commute
```

Comment:
Do you want to use the word `functions` in this sentence? Or something like "arrows" or "morphisms"?

Reply:
I mean the function here is `F.map`, but the word function could have been confusing. I improved the docs a bit to make this explicit.